src/main/kotlin/org/jitsi/nlj/transform/node/incoming/Vp8Parser.kt

```diff
@@ -60,14 +64,14 @@ class Vp8Parser(
                 numKeyframes++
             }
 
-            if (!videoRtpPacket.hasPictureId) {
-                logger.cinfo { "Packet $videoRtpPacket does not have picture ID.  Packet data: ${videoRtpPacket.toHex(80)}" }
-            } else if (!videoRtpPacket.hasExtendedPictureId) {
-                logger.cinfo { "Packet $videoRtpPacket has 7-bit (short) picture ID.  Packet data: ${videoRtpPacket.toHex(80)}" }
+            pictureIdState.setState(videoRtpPacket.hasPictureId, videoRtpPacket) {
+                "Packet Data: ${videoRtpPacket.toHex(80)}"
             }
-
-            if (videoRtpPacket.hasTemporalLayerIndex && !videoRtpPacket.hasTL0PICIDX) {
-                logger.cinfo { "Packet $videoRtpPacket has TID but does not have TL0PICIDX.  Packet data: ${videoRtpPacket.toHex(80)}" }
+            extendedPictureIdState.setState(videoRtpPacket.hasExtendedPictureId, videoRtpPacket) {
+                "Packet Data: ${videoRtpPacket.toHex(80)}"
+            }
+            tl0PicIdxWithoutTidState.setState(videoRtpPacket.hasTL0PICIDX && !videoRtpPacket.hasTemporalLayerIndex, videoRtpPacket) {
```

Comment:
Did you want the logical negation of the existing code like the ones above? It should ne `||` instead of `&&`.

Reply:
Right, I got confused about the test.